integration_test.go

```diff
@@ -0,0 +1,231 @@
+	logFilePath    = "./logfile.log"
+	logFileName    = "logfile.log"
+	configFilePath = "./fluent-bit.conf"
+)
+
+// integration test validates the end-to-end fluentbit and bigquery pipeline
+func TestPipeline(t *testing.T) {
+
+	ctx := context.Background()
+
+	//get projectID from environment
+	projectID := os.Getenv("ProjectID")
+	if projectID == "" {
+		t.Fatal("Environment variable 'ProjectID' is required to run this test, but not set currently")
+	}
+
+	// Set up BigQuery client
+	client, err := bigquery.NewClient(ctx, projectID)
+	if err != nil {
+		t.Fatalf("Failed to create BigQuery client: %v", err)
+	}
+	defer client.Close()
+
+	// Create a random dataset and table name
+	datasetHash := randString()
+	datasetID := "testdataset_" + datasetHash
+
+	tableHash := randString()
+	tableID := "testtable_" + tableHash
+
+	// Create BigQuery dataset and table in an existing project
+	tableSchema := bigquery.Schema{
+		{Name: "Message", Type: bigquery.StringFieldType},
+	}
+	dataset := client.Dataset(datasetID)
+	if err := dataset.Create(ctx, &bigquery.DatasetMetadata{Location: "US"}); err != nil {
+		t.Fatalf("Failed to create BigQuery dataset %v", err)
+	}
+	table := dataset.Table(tableID)
+	if err := table.Create(ctx, &bigquery.TableMetadata{Schema: tableSchema}); err != nil {
+		t.Fatalf("Failed to create BigQuery table: %v", err)
+	}
+
+	//Create config file with random table name
+	if err := createConfigFile(projectID, datasetID, tableID); err != nil {
+		t.Fatalf("failed to create config file: %v", err)
+	}
+
+	// Create log file
+	file, err := os.Create(logFilePath)
+	if err != nil {
+		t.Fatalf("Failed to create log file: %v", err)
+	}
+	defer file.Close()
+
+	// Start Fluent Bit with the config file
+	FBcmd := exec.Command("fluent-bit", "-c", "fluent-bit.conf")
+	if err := FBcmd.Start(); err != nil {
+		t.Fatalf("Failed to start Fluent Bit: %v", err)
+	}
+
+	//Wait for fluent-bit connection to generate data; add delays before ending fluent-bit process
+	time.Sleep(2 * time.Second)
+	if err := generateData(10); err != nil {
+		t.Fatalf("Failed to generate data: %v", err)
+	}
+	time.Sleep(2 * time.Second)
+
+	// Stop Fluent Bit
+	if err := FBcmd.Process.Kill(); err != nil {
+		t.Fatalf("Failed to stop Fluent Bit: %v", err)
+	}
+
+	// Verify data in BigQuery by querying
+	queryMsg := "SELECT Message FROM `" + projectID + "." + datasetID + "." + tableID + "`"
+	BQquery := client.Query(queryMsg)
+	BQdata, err := BQquery.Read(ctx)
+	if err != nil {
+		t.Fatalf("Failed to query data information BigQuery: %v", err)
+	}
+
+	rowCount := 0
+	for {
+		//Check that the data sent is correct
+		var BQvalues []bigquery.Value
+		err := BQdata.Next(&BQvalues)
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			t.Fatalf("Failed to read query results: %v", err)
+		}
+
+		assert.Equal(t, "hello world", BQvalues[0])
+		rowCount++
+	}
+
+	// Verify the number of rows
+	assert.Equal(t, 10, rowCount)
+
+	// Clean up - delete the BigQuery table
```

Comment:
No need to separately delete the table. However, you need to delete the dataset; that will automatically clean up the table.

Reply:
Done